Update the spec to fix the issue 782 (#943)
jakartaee/data: spec/src/main/asciidoc/method-query.asciidoc

The conditions belonging to the restriction determine the parameters of the method, as specified below in <<Query by Method Name Conditions>>.

[WARNING]
====
When using NoSQL databases, the support for restrictions varies depending on the database type:

> - **Key-Value Databases**: Support for the equals restriction and the `Not` restriction is required for the `Id` attribute. There is no requirement to support other types of restrictions, the `And` and `Or` keywords, or restrictions on other entity attributes. Queries in key-value databases are typically limited to operations using the `Id` attribute only.

Author: @njr-11 Key-value only supports `equals`. There is no way to support `Not`.

it will work like a `Map` in Java, thus, there is only the `get` to read information.

> - **Wide-Column Databases**: Wide-column databases typically support only the `AND` operator for conditions, primarily focusing on the key attribute annotated with `jakarta.nosql.Id`. Non-key attributes can be used if they are indexed as secondary indexes, which is important for compatibility, although support varies by database provider.

Reviewer: ```suggestion
- **Wide-Column Databases**: Wide-column databases must support the `AND` operator, but are not required to support the `Or` operator. Restrictions must be supported for the key attribute that is annotated with `jakarta.nosql.Id`. Support for restrictions on other attributes is not required. Typically they can be used if they are indexed as secondary indexes, although support varies by database provider.
```

> - **Wide-Column Databases**: Wide-column databases must support the `AND` operator but are not required to support the `Or` operator. Restrictions must be supported for the key attribute that is annotated with `jakarta.nosql.Id`. Support for restrictions on other attributes is not required. Typically they can be used if they are indexed as secondary indexes, although support varies by database provider.

Author: @njr-11, what do you want to do?

A) Finish those topicinat the documentation and then go to fix the TCK
B) Open a PR in parallel to fix it simultaneously.

> - **Wide-Column Databases**: Wide-column databases must support the `AND` operator but are not required to support the `Or` operator. Restrictions must be supported for the key attribute that is annotated with `jakarta.nosql.Id`. Support for restrictions on other attributes is not required. Typically they can be used if they are indexed as secondary indexes, although support varies by database provider.

Author: Thanks, it does make sense. I reviewed and moved it.

> - **Graph and Document Databases**: These databases are expected to support all the restrictions described in this section. However, precedence between `And` and `Or` operators is not guaranteed unlike relational databases. The behavior of precedence may vary significantly based on the NoSQL provider.

Reviewer: ```suggestion
- **Graph and Document Databases**: The `And` and `Or` keywords and all of the restrictions described in this section must be supported. Precedence between `And` and `Or` operators is not guaranteed and may vary significantly based on the NoSQL provider.
```

If no order is specified, the records are not sorted.

[WARNING]
====
When using NoSQL databases, sorting support varies by database type:

> - **Key-Value Databases**: Sorting is not supported for any fields. Queries are limited to operations on key attributes.

Author: `fields` I did not find a good word here; I am not sure about attribute once we are talking about the database structure, and for key, the value is a unique BLOB-like.

| `find` accepting a `PageRequest` | `Page<E>` or `CursoredPage<E>` | For use with pagination
|===

[WARNING]
====
When working with NoSQL databases, the `count` operation may not be supported, depending on the database structure and capabilities:

- **Key-Value Databases**: In general, the majority of Key-Value databases do not support counting. Their design focuses on retrieving values by keys rather than aggregating data.

> - **Other NoSQL Databases**: Support for the `count` operation varies significantly:

Reviewer: This line shouldn't be at the same level as the following two bullet points. It's also unnecessary so I'll propose omitting it
```suggestion
```

> - **Wide-Column Databases**: Counting might require additional configurations, such as enabling secondary indexes or performing costly full table scans, which can affect performance.

Reviewer: ```suggestion
- **Wide-Column Databases**: Support for count is not required. For databases that are capable of returning a count, it might require additional configuration, such as enabling secondary indexes or performing costly full table scans, which can affect performance.
```

> - **Graph and Document Databases**: These databases typically support counting, though performance may vary based on factors like indexing and query complexity.

Reviewer: ```suggestion
- **Graph and Document Databases**: Support for count is required. Performance might vary based on factors like indexing and query complexity.
```

====

> 

Author: Once we use a limited number of elements at the TCK, counting the number of elements is OK, but it is inefficient; in some databases, I would not recommend this behavior.

Plus, with a considerable amount of elements, wide-column, such as Cassandra, might make a mistake, or you need to do an `Allow filtering`` that is a linear query in multiple database instances 🐢 super slow and consumes a considerable power computer.
